walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,81 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    private void addTimeRange(List<TimeRange> timeList, int startTime, int endTime, boolean includeEnd) {
+        timeList.add((TimeRange.fromStartEnd(  
+            startTime,
+            endTime,
+            includeEnd)));
+    }
+
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        List<TimeRange> eventTimeRanges = new ArrayList<>();
+        List<TimeRange> openTimeRanges = new ArrayList<>();
+
+        if(request.getAttendees() != null && !request.getAttendees().isEmpty()) {
+            if(TimeRange.WHOLE_DAY.duration() >= request.getDuration()) {
+                for(Event loopEvents: events) {
+                    eventTimeRanges.add((loopEvents.getWhen()));
+                }
+
+                Collections.sort(eventTimeRanges, TimeRange.ORDER_BY_START);
+
+                for(int i = 0; i < eventTimeRanges.size(); i++) {
+                    if (i == 0) {
+                        addTimeRange(
+                            openTimeRanges,
+                            TimeRange.START_OF_DAY,
+                            eventTimeRanges.get(i).start(),
+                            false
+                        );
+                            if (eventTimeRanges.size() > 1) {
```

Comment:
Indentation seems a bit off, did you run this through an ide autoformatter or external linter?